src/test/java/specs/JUnitRuleExample.java

```diff
@@ -27,6 +27,16 @@
     public TemporaryFolder tempFolderRule = new TemporaryFolder();
   }
 
+  // alternative morphology of providing a rule - see http://junit.org/junit4/javadoc/4.12/org/junit/Rule.html
+  public static class TempFolderRuleProvidedViaMethodMixin {
+    private TemporaryFolder tempFolderRule = new TemporaryFolder();
+
+    @Rule
+    public TemporaryFolder getFolder() {
+      return tempFolderRule;
+    }
+  }
+
```

Comment:
This permutation of JUnit `@Rule` wasn't tested before, though supported in the code.